src/views/FacilityDetails.vue

```diff
@@ -163,29 +163,60 @@
+                  {{ translate("Add") }}
+                </ion-button>
+              </ion-card-header>
+              <ion-item v-for="store in facilityProductStores" :key="store.productStoreId">
+                <ion-label>{{ getProductStore(store.productStoreId)?.storeName || store.productStoreId }}</ion-label>
+                <ion-badge slot="end" v-if="shopifyShopIdForProductStore(store.productStoreId) !== '' && shopifyShopIdForProductStore(store.productStoreId) === current.primaryFacilityGroupId">{{ translate("primary store") }}</ion-badge>
+                <ion-button slot="end" fill="clear" color="medium" @click="productStorePopover($event, store)">
+                  <ion-icon slot="icon-only" :icon="ellipsisVerticalOutline" />
+                </ion-button>
+              </ion-item>
+              <ion-button v-if="!facilityProductStores?.length" expand="block" fill="outline" @click="selectProductStores()">
                 {{ translate("Add") }}
+                <ion-icon slot="end" :icon="addCircleOutline" />
               </ion-button>
-            </ion-card-header>
-            <ion-item v-for="store in facilityProductStores" :key="store.productStoreId">
-              <ion-label>{{ getProductStore(store.productStoreId)?.storeName || store.productStoreId }}</ion-label>
-              <ion-badge slot="end" v-if="shopifyShopIdForProductStore(store.productStoreId) !== '' && shopifyShopIdForProductStore(store.productStoreId) === current.primaryFacilityGroupId">{{ translate("primary store") }}</ion-badge>
-              <ion-button slot="end" size="default" fill="clear" color="medium" @click="productStorePopover($event, store)">
-                <ion-icon slot="icon-only" :icon="ellipsisVerticalOutline" />
-              </ion-button>
-            </ion-item>
-            <ion-button v-if="!facilityProductStores?.length" expand="block" fill="outline" @click="selectProductStores()">
-              {{ translate("Add") }}
-              <ion-icon slot="end" :icon="addCircleOutline" />
-            </ion-button>
-          </ion-card>
+            </ion-card>
+
+            <ion-card>
+              <ion-card-header>
+                <ion-card-title>
+                  {{ translate('Map Link') }}
+                </ion-card-title>
+              </ion-card-header>
+
+              <ion-card-content>
+                {{ translate('Give customers on your website a direct link to this facility on a mapping service like Google Maps.') }}
+              </ion-card-content>
+              <template v-if="contactDetails?.googleMapUrl?.infoString">
+                <ion-item lines="full" >
+                  <ion-label>{{ contactDetails.googleMapUrl.infoString }}</ion-label>
+                </ion-item>
+  
+                <ion-button fill="clear" @click="editMapUrl">
+                  {{ translate('Edit') }}
+                </ion-button>
+                <ion-button fill="clear" @click="previewMapUrl">
```

Comment:
Add href and target pattern on the button to open external link and use a computed property that returns the map url and we do not need check for http, we will assume that the url will always be a valid url.